content/docs/rfcs/13/README.md

```diff
@@ -59,139 +59,260 @@ The following are not considered as part of the adversarial model:
-
-### HistoryResponse
-
-RPC call to respond to a HistoryQuery call.
-- The `messages` field MUST contain the messages found,
-these are [14/WAKU2-MESSAGE](/spec/14) types.
-- `PagingInfo`  holds the paging information based on which the querying node can resume its further history queries. 
-  The `pageSize` indicates the number of returned Waku messages (i.e., the number of messages included in the `messages` field of `HistoryResponse`). 
-  The `direction` is the same direction as in the corresponding `HistoryQuery`. 
-  In the forward pagination, the `cursor` holds the `Index` of the last message in the `HistoryResponse` `messages` (and the first message in the backward paging). 
-  Regardless of the paging direction, the retrieved `messages` are always sorted in ascending order based on their timestamp as explained in the [sorting messages](#sorting-messages) section, that is, from the oldest to the most recent.
-  The requester SHALL embed the returned `cursor` inside its next `HistoryQuery` to retrieve the next page of the [14/WAKU2-MESSAGE](/spec/14).  
-  The `cursor` obtained from one node SHOULD NOT be used in a request to another node because the result may be different.
-- The `error` field contains information about any error that has occurred while processing the corresponding `HistoryQuery`.
-  `NONE` stands for no error.
-  This is also the default value.
-  `INVALID_CURSOR` means that the `cursor` field of `HistoryQuery` does not match with the `Index` of any of the `WakuMessage` persisted by the queried node.
+### Waku message sorting
+
+The key-value entries in the store MUST be time-sorted by the `WakuMessage` `timestamp` attribute.
+Where two or more key-value entries have identical `timestamps`,
+the entries MUST be further sorted by the natural order of their message hash keys.
+Within the context of traversing over key-value entries in the store,
+_"forward"_ indicates traversing the entries in ascending order,
+whereas _"backward"_ indicates traversing the entries in descending order.
+
+### Pagination
+
+If a large number of entries in the store service node match the query criteria provided in a `StoreRequest`,
+the client MAY make use of pagination
+in a chain of store request and response transactions
+to retrieve the full response in smaller batches termed _"pages"_.
+Pagination can be performed either in [a _forward_ or _backward_ direction](#waku-message-sorting).
+
+A store client MAY indicate the maximum number of matching entries it wants in the `StoreResponse`,
+by setting the page size limit in the `pagination_limit` field.
+Note that a store service node MAY enforce its own limit
+if the `pagination_limit` is unset
+or larger than the service node's internal page size limit.
+
+A `StoreResponse` with a populated `pagination_cursor` indicates that more stored entries match the query than included in the response.
+
+A `StoreResponse` without a populated `pagination_cursor` indicates that
```

Comment:
I a also wondering if response can include some indication showing that data older than requestedTime is purged or cleaned-up? 
Otherwise there is no way for client to know if this was the only data available for that time-period or just that this node is limited due to its configuration.
This would prompt the client to go to another storenode to get additional data.